Prusa-CW-Firmware/main.cpp

```diff
@@ -461,60 +494,58 @@ void setup() {
+  EEPROM.put(reinterpret_cast<int>(&(eeprom_base->FAN2_PREHEAT_SPEED)), FAN2_PREHEAT_SPEED);
+  EEPROM.put(reinterpret_cast<int>(&(eeprom_base->resin_preheat_run_time)), resin_preheat_run_time);
+  EEPROM.put(reinterpret_cast<int>(&(eeprom_base->resin_target_temp_celsius)), resin_target_temp_celsius);
 }
 
-
-void read_config(unsigned int address) {
+void read_config() {
   char test_magic[MAGIC_SIZE];
-  EEPROM.get(address, test_magic);
-  address += MAGIC_SIZE;
-  if (!strncmp(magic, test_magic, MAGIC_SIZE)) {
-    EEPROM.get(address++, washing_speed);
-    EEPROM.get(address++, curing_speed);
-    EEPROM.get(address++, washing_run_time);
-    EEPROM.get(address++, curing_run_time);
-    EEPROM.get(address++, finish_beep_mode);
-    EEPROM.get(address++, drying_run_time);
-    EEPROM.get(address++, sound_response);
-    EEPROM.get(address++, curing_machine_mode);
-    EEPROM.get(address++, heat_to_target_temp);
-    EEPROM.get(address++, target_temp_celsius);
-    EEPROM.get(address++, target_temp_fahrenheit);
-    EEPROM.get(address++, SI_unit_system);
-    EEPROM.get(address++, heater_failure);
-    EEPROM.get(address++, FAN1_CURING_SPEED);
-    EEPROM.get(address++, FAN1_DRYING_SPEED);
-    EEPROM.get(address++, FAN1_PREHEAT_SPEED);
-    EEPROM.get(address++, FAN2_CURING_SPEED);
-    EEPROM.get(address++, FAN2_DRYING_SPEED);
-    EEPROM.get(address++, FAN2_PREHEAT_SPEED);
-    EEPROM.get(address++, resin_preheat_run_time);
-    EEPROM.get(address++, resin_target_temp_celsius);
+  EEPROM.get(reinterpret_cast<int>(&(eeprom_base->magic)), test_magic);
+  if (!strncmp(magic2, test_magic, MAGIC_SIZE)) {
+    EEPROM.get(reinterpret_cast<int>(&(eeprom_base->washing_speed)), washing_speed);
+    EEPROM.get(reinterpret_cast<int>(&(eeprom_base->curing_speed)), curing_speed);
+    EEPROM.get(reinterpret_cast<int>(&(eeprom_base->washing_run_time)), washing_run_time);
+    EEPROM.get(reinterpret_cast<int>(&(eeprom_base->curing_run_time)), curing_run_time);
+    EEPROM.get(reinterpret_cast<int>(&(eeprom_base->finish_beep_mode)), finish_beep_mode);
+    EEPROM.get(reinterpret_cast<int>(&(eeprom_base->drying_run_time)), drying_run_time);
+    EEPROM.get(reinterpret_cast<int>(&(eeprom_base->sound_response)), sound_response);
+    EEPROM.get(reinterpret_cast<int>(&(eeprom_base->curing_machine_mode)), curing_machine_mode);
+    EEPROM.get(reinterpret_cast<int>(&(eeprom_base->heat_to_target_temp)), heat_to_target_temp);
+    EEPROM.get(reinterpret_cast<int>(&(eeprom_base->target_temp_celsius)), target_temp_celsius);
+    EEPROM.get(reinterpret_cast<int>(&(eeprom_base->target_temp_fahrenheit)), target_temp_fahrenheit);
+    EEPROM.get(reinterpret_cast<int>(&(eeprom_base->SI_unit_system)), SI_unit_system);
+    EEPROM.get(reinterpret_cast<int>(&(eeprom_base->heater_failure)), heater_failure);
+    if(!strncmp(magic, test_magic, MAGIC_SIZE)){
+    	EEPROM.get(reinterpret_cast<int>(&(eeprom_base->FAN1_CURING_SPEED)), FAN1_CURING_SPEED);
```

Comment:
It looks like logical error there. This path of code will never be executed, as test_magic will never be equal to both magic and magic2.

Reply:
It looks OK now.